### PR TITLE
Add model builder for jit test

### PR DESCRIPTION
**Pull request by nishantpdce**
Summary: Add model builder file to create simple models for testing jit passes

Differential Revision: D14537862

### CODE REVIEW

**Review by apaszke — CHANGES_REQUESTED** [review]
Currently the file does nothing at the moment. What's the plan for this test?

**Comment by nishantpdce** [comment]
> Currently the file does nothing at the moment. What's the plan for this test?

There is a follow up change which will be using this change for jit passes

**Comment by ZolotukhinM** [comment]
> Currently the file does nothing at the moment. What's the plan for this test?

Think about this file as a collection of convenience methods for creating similar to real-world models. The plan is to build kind of integration testing on top of it (along with small local tests for each particular jit pass). I'm not sure we need it right now, but I can see how it could be useful in future.

File: test/model_builder.py
Comment (ZolotukhinM) [inline test/model_builder.py]: This will fail if `model_name` != `None` is passed.

**Review by ZolotukhinM — COMMENTED** [review]
It looks good to me (modulo one remark inline), but I'd wait for @apaszke's approval since originally he had questions about this.

**Review by apaszke — CHANGES_REQUESTED** [review]
>  I'm not sure we need it right now, but I can see how it could be useful in future.

I understand that this _might_ be useful at some point, but that's not a good enough reason to check in code. If it _doesn't_ turn out to be useful it will stay there unused and will only pollute the test directory. It's ok to merge this, but only once it is used _somewhere_.

**Pull request closed without merging**
